editor/src/core/model/element-metadata-utils.ts

```diff
@@ -2099,6 +2096,41 @@ function fillSpyOnlyMetadata(
     }
   })
 
+  const elementsWithoutGlobalContentBox = Object.keys(fromSpy).filter((p) => {
+    return fromDOM[p]?.specialSizeMeasurements.globalContentBoxForChildren == null
+  })
+
+  // sorted, so that parents are fixed first
+  elementsWithoutGlobalContentBox.sort()
+
+  fastForEach(elementsWithoutGlobalContentBox, (pathStr) => {
+    const spyElem = fromSpy[pathStr] ?? conditionalsWithDefaultMetadata[pathStr]
+    const domElem = fromDOM[pathStr]
+    const sameThingFromWorkingElems = workingElements[pathStr]
+
+    const parentPathStr = EP.toString(EP.parentPath(EP.fromString(pathStr)))
+
+    const parentGlobalContentBoxForChildren =
+      fromDOM[parentPathStr]?.specialSizeMeasurements.globalContentBoxForChildren ??
+      workingElements[parentPathStr]?.specialSizeMeasurements.globalContentBoxForChildren ??
+      infinityCanvasRectangle
+
+    workingElements[pathStr] = {
+      ...spyElem,
+      ...sameThingFromWorkingElems,
```

Comment:
can't you just merge it on the `...sameThingFromWorkingElems` here, leaving out the spyElem completely?